nixos/modules/services/misc/snapper.nix

```diff
@@ -3,7 +3,101 @@
 with lib;
 
 let
+  inherit (lib.strings) concatMapStrings concatMapStringsSep escape removeSuffix toUpper;
+
   cfg = config.services.snapper;
+
+  camelToCaps = with builtins; s: toUpper (concatMapStrings (p: if isList p then "${elemAt p 0}_${elemAt p 1}" else p) (split "([^A-Z])([A-Z])" s));
```

Comment:
What does this abstraction buy us, aside from confusion? I would recommend we rename the existing options to follow upstream config and put less abstraction between us and upstream. We can directly reference upstream documentation that way and not ask users to do mental conversion in their head that we simply undo.

Reply:
Is your recommendation that the documented options also use SHOUTING_CAPS?

If so, the existing `subvolume` and `fstype` would then want to be renamed to `SUBVOLUME` and `FSTYPE`, which I am happy to do; should I leave in support for the old names with some sort of deprecation warning? Or refrain from renaming them and let them stick around as exceptions?

If not, there's an awkward tension between documented and undocumented options which would arise any time someone wanted to promote an option from the latter to the former, which doesn't seem great.

Reply:
Also, what would be the best way to represent the renaming of `services.snapper.configs.<name>.subvolume` to `services.snapper.configs.<name>.SUBVOLUME`? I don't think `mkRenamedOptionModule` will work because the path has that dynamic `<name>` part in it, and `SUBVOLUME` should be a required option. I could give it a `nullOr` type and a `default = null;`, and manually throw an error if neither `subvolume` nor `SUBVOLUME` is present and non-null. Is there a better approach?

Reply:
> We have `mkRenamedOption` for this.

@SuperSandro2000, I have only found `mkRenamedOptionModule`, which as far as I can tell expects the full attribute paths in question to be statically known. How would I use it with `services.snapper.configs.<name>.subvolume`, where the `<name>` part is variable?

Reply:
For some reason I thought this functionality was added to NixOS... but after looking it appears it is not.

What you can do in this case is keep the option around but don't advertise it (you can add `internal = true;`, set type to `nullOr ...`) and then add `warnings` or `assertions` if someone uses the option.

Reply:
Since this module was basically useless without users defining `extraConfig`, if we're dropping support for `extraConfig` almost all users will need to migrate their configs anyway. In which case, I don't really see any point in supporting the legacy names, so I didn't keep options for them around, merely added assertions that check for them.